src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -137,23 +140,58 @@ def __init__(self, start_msg='', finish_msg='',
     def _delay(self):
         time.sleep(self.poller_done_interval_ms / 1000.0)
 
+    def _template_progress(self, correlation_id):
+        """ gets the progress for template deployments """
+        if correlation_id is not None:
+
+            formatter = "eventTimestamp ge {} and eventTimestamp le {}"
+
+            end_time = datetime.datetime.utcnow()
+            start_time = end_time - datetime.timedelta(seconds=DEFAULT_QUERY_TIME_RANGE)
+            odata_filters = formatter.format(start_time.strftime('%Y-%m-%dT%H:%M:%SZ'),
+                                             end_time.strftime('%Y-%m-%dT%H:%M:%SZ'))
+
+            odata_filters = "{} and {} eq '{}'".format(odata_filters, 'correlationId', correlation_id)
+
+            from azure.cli.core.commands.client_factory import get_mgmt_service_client
+            from azure.monitor import MonitorClient
```

Comment:
These imports should go at the top of the function.
Import in the middle of a function isn't clean.